rpc/lib/types/types.go

```diff
@@ -13,17 +13,63 @@ import (
 	tmpubsub "github.com/tendermint/tendermint/libs/pubsub"
 )
 
+// a wrapper to emulate a sum type: jsonrpcid = string | int
+// TODO: refactor when Go 2.0 arrives https://github.com/golang/go/issues/19412
```

Comment:
LOL

Reply:
Yeah, it's a bit unfortunate accidental complexity